VEDA_OS01/models.py

```diff
@@ -40,6 +41,111 @@ class TranscriptStatus(object):
     )
 
 
+class Cielo24Turnaround(object):
+    """
+    Cielo24 turnarounds.
+    """
+    STANDARD = 'STANDARD'
+    PRIORITY = 'PRIORITY'
+    CHOICES = (
+        (STANDARD, 'Standard, 48h'),
+        (PRIORITY, 'Priority, 24h'),
+    )
+
+
+class Cielo24Fidelity(object):
+    """
+    Cielo24 fidelity.
+    """
+    MECHANICAL = 'MECHANICAL'
+    PREMIUM = 'PREMIUM'
+    PROFESSIONAL = 'PROFESSIONAL'
+    CHOICES = (
+        (MECHANICAL, 'Mechanical, 75% Accuracy'),
+        (PREMIUM, 'Premium, 95% Accuracy'),
+        (PROFESSIONAL, 'Professional, 99% Accuracy'),
+    )
+
+
+class ThreePlayTurnaround(object):
+    """
+    3PlayMedia turnarounds.
+    """
+    EXTENDED_SERVICE = 'extended_service'
+    DEFAULT = 'default'
+    EXPEDITED_SERVICE = 'expedited_service'
+    RUSH_SERVICE = 'rush_service'
+    SAME_DAY_SERVICE = 'same_day_service'
+
+    CHOICES = (
+        (EXTENDED_SERVICE, '10-Day/Extended'),
+        (DEFAULT, '4-Day/Default'),
+        (EXPEDITED_SERVICE, '2-Day/Expedited'),
+        (RUSH_SERVICE, '24 hour/Rush'),
+        (SAME_DAY_SERVICE, 'Same Day'),
+    )
+
+
+class ListField(models.TextField):
+    """
+    A List Field which can be used to store and retrieve pythonic list of strings.
+    """
+    def get_prep_value(self, value):
+        """
+        Converts a list to its json representation to store in database as text.
+        """
+        if value and not isinstance(value, list):
+            raise ValueError(u'The given value {} is not a list.'.format(value))
+
+        return json.dumps(self.validate_list(value) or [])
+
+    def from_db_value(self, value, expression, connection, context):
+        """
+        Converts a json list representation in a database to a python object.
+        """
+        return self.to_python(value)
+
+    def to_python(self, value):
+        """
+        Converts the value into a list.
+        """
+        if not value:
+            value = []
+
+        # If a list is set then validated its items
+        if isinstance(value, list):
+            py_list = self.validate_list(value)
+        else:  # try to de-serialize value and expect list and then validate
+            try:
+                py_list = json.loads(value)
+                if not isinstance(py_list, list):
+                    raise TypeError
+
+                self.validate_list(py_list)
+            except (ValueError, TypeError):
+                raise ValueError(u'Must be a valid list of strings.')
+
+        return py_list
+
+    def validate_list(self, value):
+        """
+        Validate the data before saving into the database.
+
+        Arguments:
+            value(list): list to be validated
+
+        Returns:
+            A list if validation is successful
+
+        Raises:
+            ValidationError
+        """
+        if all(isinstance(item, basestring) for item in value) is False:
+            raise ValueError(u'list must only contain strings.')
+
+        return value
```

Comment:
I'm assuming this code is already being reviewed and tested. Let me know if that's not the case.